File: leetcode/java/1672. Richest Customer Wealth.java
class Solution {
    public int maximumWealth(int[][] accounts) {
        
        final int rows = accounts.length;
        final int cols = accounts[0].length;
        
        int ret = 0;
        int sum = 0;
        for(int i = 0; i < rows; i++) {
            
            sum = 0;
            
            for(int j = 0; j < cols; j++) {
                sum += accounts[i][j];
            }
            
            ret = (sum > ret) ? sum : ret;
        }
        
        return ret;
    }
}

File: leetcode/kotlin/1672. Richest Customer Wealth.kt
class Solution {
    fun maximumWealth(accounts: Array<IntArray>): Int {
        
        // leetcode 내 kotlin 버전이 낮아서 그런건지 해당 함수는 작동 안 됨
        // return accounts.map {
        //     it.sum()
        // }.maxOrNull() ?: 0
        
        val sums = accounts.map {
            it.sum()
        }
        
        var ret = -1
        sums.forEach {
            ret = if(it > ret) it else ret
        }
        
        return ret
    }
}
